### PR TITLE
W-13829786: Avoid reading mule-module.properties for modules with module-info

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: modules/container/src/main/java/org/mule/runtime/container/internal/ContainerModulesDiscovererProvider.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.container.internal;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ContainerModulesDiscovererProvider {
```
Comment (asanguinetti) [inline modules/container/src/main/java/org/mule/runtime/container/internal/ContainerModulesDiscovererProvider.java:12]: JDoc

File: modules/container/src/main/java11/org/mule/runtime/container/internal/ContainerModulesDiscovererProvider.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.container.internal;
+
+import static org.mule.runtime.api.util.MuleSystemProperties.classloaderContainerJpmsModuleLayer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ContainerModulesDiscovererProvider {
```
Comment (asanguinetti) [inline modules/container/src/main/java11/org/mule/runtime/container/internal/ContainerModulesDiscovererProvider.java:14]: JDoc

File: tests/infrastructure/src/main/java/org/mule/test/infrastructure/deployment/AbstractFakeMuleServerTestCase.java [hunk 3]
```diff
@@ -35,6 +36,10 @@ public class AbstractFakeMuleServerTestCase extends AbstractMuleTestCase {
   public SystemProperty jvmVersionExtensionEnforcementLoose =
       new SystemProperty("mule.jvm.version.extension.enforcement", "LOOSE");
 
+  @Rule
+  public SystemProperty classloaderContainerJpmsModuleLayer =
+      new SystemProperty(CLASSLOADER_CONTAINER_JPMS_MODULE_LAYER, "false");
```
Comment (asanguinetti) [inline tests/infrastructure/src/main/java/org/mule/test/infrastructure/deployment/AbstractFakeMuleServerTestCase.java:41]: Why is it that we are enforcing this for these tests? Could you elaborate?

**Review by asanguinetti — CHANGES_REQUESTED** [review]

**Review by asanguinetti — APPROVED** [review]

**Comment by elrodro83** [comment]
--validate

**Pull request merged**